pkg/virt-controller/watch/vmi.go

```diff
@@ -1612,19 +1611,33 @@ func (c *VMIController) handleHotplugVolumes(hotplugVolumes []*virtv1.Volume, ho
 		}
 	}
 
-	if len(currentPod) == 0 && len(readyHotplugVolumes) > 0 {
-		// ready volumes have changed
-		// Create new attachment pod that holds all the ready volumes
-		if err := c.createAttachmentPod(vmi, virtLauncherPod, readyHotplugVolumes); err != nil {
-			return err
+	switch len(currentPod) {
```

Comment:
Any reason you used a switch instead of if len(currentPod) == 0 { ... } else { ... }